SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebuggerEditor.cpp

```diff
@@ -0,0 +1,158 @@
+#include "Utils/SpatialDebuggerEditor.h"
+
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/InspectionColors.h"
+#include "Debug/DebugDrawService.h"
+#include "Editor.h"
+#include "LevelEditor.h"
+#include "Modules/ModuleManager.h"
+#include "Engine/World.h"
+#include "GameFramework/WorldSettings.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+
+using namespace SpatialGDK;
+
+ASpatialDebuggerEditor::ASpatialDebuggerEditor(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = false;
+	PrimaryActorTick.bStartWithTickEnabled = false;
+	PrimaryActorTick.TickInterval = 1.f;
+
+	bAlwaysRelevant = false;
+	bNetLoadOnClient = false;
+	bReplicates = false;
+
+	NetUpdateFrequency = 1.f;
+
+	bShowWorkerRegions = false;
+	bIsGameInProgress = false;
+
+	OnBeginPieHandle = FEditorDelegates::BeginPIE.AddUObject(this, &ASpatialDebuggerEditor::OnPieBeginEvent);
+	OnEndPieHandle = FEditorDelegates::EndPIE.AddUObject(this, &ASpatialDebuggerEditor::OnPieEndEvent);
+}
+
+void ASpatialDebuggerEditor::BeginPlay()
+{
+	// Override parent to do nothing
```

Comment:
Is there a reason to override here? Is this even called since it isn't on at runtime?  If it's not called, I'd say don't override it.

If it is called and you need to override it, add a call to AActor::BeginPlay() otherwise you're bypassing the whole AActor initialization.

Reply:
This is not called so I have removed it.